### DISCUSSION TITLE
Import the world with Flatnode file

### DISCUSSION THREAD

**Discussion by gsg-git** [mediagis/nominatim-docker#481] [Q&A] [answered]
Since my other container seems ruined i rather ask in a new Discussion.

I want to import the world and this is what i want to use ant want to make shure everything ist correct bevore i ruin another container and time.
Is the flatnode file correct and will it be generated like this or do i have to do something after the docker run skript?

The device has 14 Cores(i5-13600) and 128GB RAM
This is the Command i plan to use:

docker run -it \
	--shm-size=60g \
	-e THREADS=12 \
	-e NOMINATIM_FLATNODE_FILE=/nominatim/flatnode/flatnode.file \
	-e POSTGRES_SHARED_BUFFERS=2GB \
	-e POSTGRES_MAINTAINENCE_WORK_MEM=10GB \
	-e POSTGRES_AUTOVACUUM_WORK_MEM=2GB \
	-e POSTGRES_WORK_MEM=50MB \
	-e POSTGRES_EFFECTIVE_CACHE_SIZE=24GB \
	-e POSTGRES_SYNCHRONOUS_COMMIT=off \
	-e POSTGRES_MAX_WAL_SIZE=1GB \
	-e POSTGRES_CHECKPOINT_TIMEOUT=10min \
	-e POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9 \
	-e PBF_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/pbf/planet-latest.osm.pbf \
	-e REPLICATION_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/replication/day/ \
	-e IMPORT_WIKIPEDIA=true \
	-p 8080:8080 \
	--name nominatim \
	mediagis/nominatim:4.2

**Comment by leonardehrenfried** [comment]
No, it should just be `-v nominatim-flatnode:/nominatim/flatnode`

I would love it if you send a documentation PR.

**Reply by gsg-git** [reply]
Thank you!


then:
docker run -it \
	--shm-size=60g \
	-e THREADS=10 \
	-v NOMINATIM_FLATNODE_FILE=/nominatim/flatnode/flatnode.file \
	-e POSTGRES_SHARED_BUFFERS=2GB \
	-e POSTGRES_MAINTAINENCE_WORK_MEM=10GB \
	-e POSTGRES_AUTOVACUUM_WORK_MEM=2GB \
	-e POSTGRES_WORK_MEM=50MB \
	-e POSTGRES_EFFECTIVE_CACHE_SIZE=24GB \
	-e POSTGRES_SYNCHRONOUS_COMMIT=off \
	-e POSTGRES_MAX_WAL_SIZE=1GB \
	-e POSTGRES_CHECKPOINT_TIMEOUT=10min \
	-e POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9 \
	-e PBF_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/pbf/planet-latest.osm.pbf \
	-e REPLICATION_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/replication/day/ \
	-e IMPORT_WIKIPEDIA=true \
	-p 8080:8080 \
	--name nominatim \
	mediagis/nominatim:4.2

**Reply by gsg-git** [reply]
i will send you a mail with my journey report.

**Reply by leonardehrenfried** [reply, marked as answer]
Your example is wrong.

Please read my comment again and replace the line about the flatnode file with `-v nominatim-flatnode:/nominatim/flatnode`.

BTW, sending a "documentation PR" means a Github pull request not a personal email.

**Reply by gsg-git** [reply]
okay i see the error - i copied it from other [discussions](https://github.com/mediagis/nominatim-docker/discussions/280)
Did the syntax cange until today?
i have no idea how to make the pull request - i have to look it up. I made a suggestion [here:](https://github.com/mediagis/nominatim-docker/discussions/479)




docker run -it \
	--shm-size=60g \
	-e THREADS=10 \
	-v nominatim-flatnode:/nominatim/flatnode \
	-e POSTGRES_SHARED_BUFFERS=2GB \
	-e POSTGRES_MAINTAINENCE_WORK_MEM=10GB \
	-e POSTGRES_AUTOVACUUM_WORK_MEM=2GB \
	-e POSTGRES_WORK_MEM=50MB \
	-e POSTGRES_EFFECTIVE_CACHE_SIZE=24GB \
	-e POSTGRES_SYNCHRONOUS_COMMIT=off \
	-e POSTGRES_MAX_WAL_SIZE=1GB \
	-e POSTGRES_CHECKPOINT_TIMEOUT=10min \
	-e POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9 \
	-e PBF_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/pbf/planet-latest.osm.pbf \
	-e REPLICATION_URL=https://ftp5.gwdg.de/pub/misc/openstreetmap/planet.openstreetmap.org/replication/day/ \
	-e IMPORT_WIKIPEDIA=true \
	-p 8080:8080 \
	--name nominatim \
	mediagis/nominatim:4.2

**Reply by gsg-git** [reply]
Whish me luck: the process has been started.
![grafik](https://github.com/mediagis/nominatim-docker/assets/92863111/ebaeca19-3e1a-4e9c-8c00-d1c179810a04)

**Reply by gsg-git** [reply]
@leonardehrenfried 
The Container is now up and running!
Thank you!
